### PR TITLE
Audio: add buffer (un)queuing to Source.

### PR DESCRIPTION
**Pull request by williamjcm**
And also add an implementation for Source::type().

Reviews and feedback are welcome!

### CODE REVIEW

File: src/Magnum/Audio/Source.cpp [hunk 1]
```diff
@@ -38,6 +41,32 @@ Source& Source::setBuffer(Buffer* buffer) {
...
+
+std::size_t Source::unqueueBuffers(Containers::ArrayView<Containers::Reference<Buffer>> buffers) {
+    ALint processedBuffers;
+    alGetSourcei(_id, AL_BUFFERS_PROCESSED, &processedBuffers);
+
+    if(!processedBuffers) return 0;
+
+    Containers::Array<ALuint> unqueuedIds(processedBuffers);
+    alSourceUnqueueBuffers(_id, unqueuedIds.size(), unqueuedIds.data());
+    auto isNotUnqueued = [&unqueuedIds](Containers::Reference<Buffer> buffer){
```
Comment (mosra) [inline src/Magnum/Audio/Source.cpp]: would the following work too?

```suggestion
    auto isNotUnqueued = [&unqueuedIds](Buffer& buffer) {
```

File: src/Magnum/Audio/Source.h [hunk 2]
```diff
@@ -734,6 +758,12 @@ inline Source& Source::operator=(Source&& other) {
     return *this;
 }
 
+inline auto Source::type() const -> Type {
```
Comment (mosra) [inline src/Magnum/Audio/Source.h]: As we discussed on Gitter, can you add a simple test for this? Thanks!

Comment (williamjcm) [reply src/Magnum/Audio/Source.h]: I forgot to push it. 🤦‍♂

File: src/Magnum/Audio/Source.cpp (hunk 1 above)
Comment (williamjcm) [reply src/Magnum/Audio/Source.cpp]: I forgot that `Reference<Type>` was convertible to `Type&`. 🤦‍♂

**Comment by mosra** [comment]
Merged in b9fb03ab8e30e52e66bd3da6ab5deefff42aabfe and 1104d8bb15cfb42e89aa89d1248d2083239e89eb, thank you! While at it, in c442c07190c010f854a3f5e1216f4e7e26b5773b I added also debug output for `Source::Type`.

**Pull request closed without merging**